app/helpers/sources_helper.rb

```diff
@@ -13,4 +13,36 @@ def render_media_asset
       render partial: 'video'
     end
   end
+
+  ##
+  # Render a back link to the referring page if the referring page is a guide or
+  # set.  Links from guides or sets to sources may be created by admins, so the
+  # applicaiton cannot expect that back URI's can be included in params.
+  # Therefore, it is necessary to parse the referring URI to see if it is a
+  # guide or set, and render a back link accordingly.
+  #
+  # @return [String] HTML back link
+  def render_back_link
+    return unless request.referer.present?
+    uri = URI(request.referer)
+    # return if referrer is an offsite link
+    return unless uri.host == request.host
+
+    uri_path = uri.path
+    # return if referrer does not have relative url root
+    return unless uri_path.slice!(Settings.relative_url_root).present?
```

Comment:
`present?` is a, as you suspected, Rails method - here's the documentation: http://apidock.com/rails/v3.1.0/Object/present%3F

I didn't use `#start_with` just b/c I was trying to be efficient - I have to do the `slice!` anyway b/c I want to remove the relative root from `uri_path` before I do `Rails.application.routes.recognize_path(uri_path)` in the next line.  But, I get what you're saying that it's hard to read and understand what's going on.  Do you think it would be better if I did:

    return unless uri_path.start_with?(Settings.relative_url_root)
    uri_path.slice!(Settings.relative_url_root)